x-pack/test/alerting_api_integration/spaces_only/tests/alerting/event_log_alerts.ts

```diff
@@ -69,6 +69,18 @@ export default function eventLogAlertTests({ getService }: FtrProviderContext) {
         (event: IValidatedEvent) => event?.event?.action !== 'execute'
       );
 
+      // Verify unique executionId generated per `action:execute` grouping
+      const eventExecutionIdSet = new Set();
+      events.forEach((event) => {
+        if (event?.event?.action === 'execute') {
+          eventExecutionIdSet.add(event?.kibana?.alert?.rule?.execution?.uuid);
+          expect(eventExecutionIdSet.size).to.equal(1);
+          eventExecutionIdSet.clear();
+        } else {
+          eventExecutionIdSet.add(event?.kibana?.alert?.rule?.execution?.uuid);
+        }
+      });
+
```

Comment:
From your [comment here](https://github.com/elastic/kibana/pull/113058#issuecomment-996016274) @pmuellr, this should cover the scenarios where new `executionId`'s should only be generated after a subsequent `action:execute`'s if I'm understanding correctly.

i.e. Any action after `action:execute` should generate a new `execution.uuid`, but it should not change between subsequent other actions (e.g. `action:new-instance` then `action:active-instance`).

Below is what an example test run grouping would look like. If this _isn't correct_, that means I've got incorrect logic over in the alerting/server/task_manager and will need to fix that.

![image](https://user-images.githubusercontent.com/2946766/149600979-f70cb981-450d-4c87-b057-f20c22eacece.png)

Reply:
I'll wait for @pmuellr to validate, but this looks right to me.

It feels like we should add another test case along side this one that validates that the same `uuid` appears on multiple alerts detected by the same rule execution. 🤔

Reply:
I'll have to look closer at this - but I believe you are correct.  But the assumption "ordering" with the notion of "after".  This is all end up based on the timestamp, I believe (from the EL query), so there probably is some kind of race condition that if the rule runs again, before processing the events, that you'd have more of a mish-mash.  Seems **very** unlikely though.

Reply:
That logic seems fine for now.  

> It feels like we should add another test case along side this one that validates that the same uuid appears on multiple alerts detected by the same rule execution

Not sure about that - doesn't it already do that?  Or it should make sure that there's at least > 1 *-instance events?

Reply:
It looks like this validates that eac execution has a unique ID, but it doesn't validate that a single execution with multiple alerts end up with the same unique id - which they should.
Feels like we're missing a test case. 🤔

Reply:
Looking at it again, I think the code is already checking that all the events have the same execution id.  I guess there's a question if there are > 1 *-instance events - if there's only one *-instance event, is that checking good enough?  Seems like it to me.

One thing that would be interesting to check is if ALL of the *-instance methods are actually covered.  We could just keep a counter over the `event.action` values, make sure it's > 0 for new- active- and recovered- instance actions.

The other thing we don't check, is that different executions have different id's.  We could keep a Set for the execution id's for the execute events, and then make sure that size === # of execution events.

Reply:
>Looking at it again, I think the code is already checking that all the events have the same execution id. I guess there's a question if there are > 1 *-instance events - if there's only one *-instance event, is that checking good enough? Seems like it to me.

Oh, this might be a misunderstanding on my part with regards to the Event Log `event.action` values.
In Garret's example output I can see two lines with the same UUID - `active-instance` and `new-instance`, are those two **different** alerts? Or the same one?
If different then you're 100% right... I think I might have been confusing event names. 🤔

Reply:
> The other thing we don't check, is that different executions have different id's. We could keep a Set for the execution id's for the execute events, and then make sure that size === # of execution events.

Thanks for discussing and verifying over zoom @pmuellr! Remaining coverage for the above added in a15d328 🙂